src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,407 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "application_manager/help_prompt_manager_impl.h"
+#include "application_manager/application.h"
+#include "application_manager/application_manager.h"
+#include "application_manager/commands/command_impl.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/smart_object_keys.h"
+#include "smart_objects/smart_object.h"
+#include "utils/logger.h"
+#include "utils/make_shared.h"
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "HelpPromptManagerImpl")
+
+namespace {
+const std::size_t kLimitCommand = 30;
+}
+
+namespace application_manager {
+
+HelpPromptManagerImpl::HelpPromptManagerImpl(Application& app,
+                                             ApplicationManager& app_manager)
+    : app_(app)
+    , app_manager_(app_manager)
+    , sending_type_(SendingType::kSendBoth)
+    , count_requests_commands_(0)
+    , is_tts_send_(false)
+    , is_ui_send_(false) {}
+
+HelpPromptManagerImpl::~HelpPromptManagerImpl() {
+  LOG4CXX_AUTO_TRACE(logger_);
+}
+
+bool HelpPromptManagerImpl::AddCommand(
+    const uint32_t cmd_id, const smart_objects::SmartObject& command) {
+  sync_primitives::AutoLock lock(vr_commands_lock_);
+  auto it = vr_commands_.find(cmd_id);
+  if (vr_commands_.end() != it) {
+    LOG4CXX_DEBUG(logger_, "Commands with id:" << cmd_id << " already exists");
+    return false;
+  }
+
+  if (false == command.keyExists(strings::vr_commands)) {
+    LOG4CXX_DEBUG(logger_, "vr_commands does`t present");
+    return false;
+  }
+  const smart_objects::SmartObject& commands = command[strings::vr_commands];
+
+  const std::size_t limit = kLimitCommand - count_requests_commands_;
```

Comment:
No need for `limit` variable, should just be comparing `count_requests_commands_` and `kLimitCommand`.

Reply:
Better to rename `count_requests_commands_` to `count_vr_commands`

Reply:
@jacobkeeler agree here. `limit` variable removed at this point. Also counter variable was renamed. See ab21400